src/sentry/debug_files/artifact_bundle_indexing.py

```diff
@@ -468,13 +481,31 @@ def to_json(self) -> str:
             for bundle in self._bundles
         ]
         json_idx: Dict[str, Any] = {
+            "is_complete": self._is_complete,
             "bundles": bundles,
             "files_by_url": self._files_by_url,
             "files_by_debug_id": self._files_by_debug_id,
         }
 
         return json.dumps(json_idx)
 
+    def enforce_size_limits(self):
+        """
+        This enforced reasonable limits on the data we put into the `FlatFileIndex` by removing
+        the oldest bundle from the index until the limits are met.
+        """
+        bundles_by_timestamp = [bundle for bundle in self._bundles]
+        bundles_by_timestamp.sort(reverse=True, key=lambda bundle: (bundle.timestamp, bundle.id))
+
+        while (
+            len(self._bundles) > MAX_BUNDLES_PER_INDEX
+            or len(self._files_by_debug_id) > MAX_DEBUGIDS_PER_INDEX
+            or len(self._files_by_url) > MAX_URLS_PER_INDEX
+        ):
+            bundle_to_remove = bundles_by_timestamp.pop()
```

Comment:
Can we maybe track how many bundles we are removing? I think it's important to understand if our system is already reaching its upper bounds.